internal/server/server.go

```diff
@@ -310,18 +310,28 @@ func (s *Server) getPostgresConnectionString() (string, error) {
 
 	if s.options.DBHost != "" {
 		// config has separate postgres parameters set, combine them into a connection DSN now
-		fmt.Fprintf(&pgConn, "host=%s ", s.options.DBHost)
+		host, err := secrets.GetSecret(s.options.DBHost, s.secrets)
```

Comment:
As Bruce was saying, I don't quite understand why dbhost, username, and dbname would be secrets. We can already read these values from environment variables.

If we wanted them to also come from files we could use `types.StringOrFile`, but I don't know why we'd want them to come from files.

Even if you use a kube secret to store them, can't we set values from [a kube secret as environment variables on the pod](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)?